web/src/lib/data-hooks/useBusinessNameSearch.ts

```diff
@@ -0,0 +1,104 @@
+import * as api from "@/lib/api-client/apiClient";
+import { useUserData } from "@/lib/data-hooks/useUserData";
+import { NameAvailability, SearchBusinessNameError } from "@/lib/types/types";
+import analytics from "@/lib/utils/analytics";
+import { ChangeEvent, FocusEvent, FormEvent, useState } from "react";
+
+export const useBusinessNameSearch = (
```

Comment:
Abstracted out the state and fetching logic into a custom hook here. This felt better than the two business name search's sharing the same React component, as there were lots of subtle differences in the display logic, it felt cleaner to have shared hook and separate UI components.